From b9edcca1f41a942ddfec59974f74e67d9e9e59ab Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Sat, 2 Dec 2017 12:58:11 +0000
Subject: [PATCH 1/1] linux: add 'StartupWMClass' to desktop file

This fixes various system integration things for when WM_CLASS and
WM_NAME don't match, some desktop enviroments handle this
better than others. This fixes:

* The application appearing twice in docks
* No pin to panel option shown in Budgie Desktop
* The themed icon not being used in certain situations.

Signed-off-by: Joey Riches <josephriches@gmail.com>
---
 linux/applications/org.qupzilla.QupZilla.desktop | 1 +
 1 file changed, 1 insertion(+)

diff --git a/linux/applications/org.qupzilla.QupZilla.desktop b/linux/applications/org.qupzilla.QupZilla.desktop
index 7299fb17..a97e9d07 100644
--- a/linux/applications/org.qupzilla.QupZilla.desktop
+++ b/linux/applications/org.qupzilla.QupZilla.desktop
@@ -78,6 +78,7 @@ GenericName[fa]=مرورگر وب
 GenericName[ar]=متصفح وِب
 GenericName[fi]=WWW-selain
 Exec=qupzilla %u
+StartupWMClass=QupZilla
 MimeType=text/html;application/xhtml+xml;x-scheme-handler/http;x-scheme-handler/https;x-scheme-handler/ftp;application/x-mimearchive;
 Terminal=false
 Actions=NewTab;NewWindow;PrivateBrowsing;
-- 
2.15.1

